snapcraft/internal/meta/_snap_packaging.py

```diff
@@ -438,36 +438,42 @@ def _generate_snapcraft_runner(self) -> Optional[str]:
 
         Return path relative to prime directory, if created."""
 
+        # If there are no apps, no need to create a runner.
+        if not self._snap_meta.apps:
```

Comment:
can you take the snapd snap type into account as well?

Reply:
Done